### PR TITLE
Remove semconv

### PR DESCRIPTION
**Pull request by jack-berg**
Not sure if we should remove right away or give the deprecation notice a few releases to sink in. Opening the PR just to have it done and to prompt the discussion.

### CODE REVIEW

**Review by mateuszrzeszutek — APPROVED** [review]
IMO it should be fine to remove that. There's a link to the new repo in the README, it's discoverable, I don't think there's anything more that we need to do.

File: README.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ write your own **manual instrumentation**, or how to set up the OpenTelemetry
 Java SDK, see [Manual instrumentation][]. Fully-functional examples
 are available in [opentelemetry-java-docs][].
 
+If you are looking for generated classes for
+the [OpenTelemetry semantic conventions][opentelemetry-semantic-conventions],
+see [semantic-conventions-java][opentelemetry-semantic-conventions-java].
```
Comment (breedx-splk) [inline README.md:22]: 👍🏻

**Review by breedx-splk — APPROVED** [review]
😎

**Comment by jack-berg** [comment]
Should resolve #5849 before merging.

**Comment by jack-berg** [comment]
Anyone following please see this [comment](https://github.com/open-telemetry/opentelemetry-java/issues/5849#issuecomment-1736258339). I think we should proceed merging this without doing the maven relocation strategy.

**Comment by trask** [comment]
> Anyone following please see this [comment](https://github.com/open-telemetry/opentelemetry-java/issues/5849#issuecomment-1736258339). I think we should proceed merging this without doing the maven relocation strategy.

makes sense, thanks for writing up your analysis!

**Pull request merged**